scripts/plot_gpx.py

```diff
@@ -0,0 +1,109 @@
+from parse_tracks import getTrackData
+from bs4 import BeautifulSoup
+import os
+import argparse
+import copy
+import sys
+import shutil
+
+sys.setrecursionlimit(15000)
+
+parser = argparse.ArgumentParser(description='Script to plot gpx data on pptx')
+parser.add_argument('--unpack_path', help='Path to sample unpack pptx file', required=True)
+parser.add_argument('--tracks_path', help='Path to gpx tracks file', required=True)
+
+args = parser.parse_args()
+
+unpack_path = args.unpack_path
+tracks_path = args.tracks_path
+
+temps = tracks_path.split("/")
+ppt_name = temps[len(temps)-1]
+
+temp_unpack_path = ppt_name.split(".")[0]+"_temp"
+shutil.copytree(unpack_path, temp_unpack_path)
+
+slide_path = temp_unpack_path+"/ppt/slides/slide1.xml"
+
+def createPptxFromTrackData(trackData):
+    soup = BeautifulSoup(open(slide_path, 'r').read(), 'xml')
+    shape_tag = soup.find('sp')
+    shape_tag.extract()
+    trackCount = 0
+    for track in trackData:
+        temp_shape_tag = copy.deepcopy(shape_tag)
+
+        path_tag = temp_shape_tag.find('path')
+        # if(trackCount!=0):
+        #     temp_shape_tag.find('off')['x'] = str(int(temp_shape_tag.find('off')['x']) + 2000000)
+            # temp_shape_tag.find('off')['y'] = str(int(temp_shape_tag.find('off')['y']) + 2000000)
+        # 
+        # w = path_tag['w']
+        # h = path_tag['h']
+
+        #Adding coordinates
+        coordinates_detail = track['coordinates']
+        coordinates = []
+        for coordinate_detail in coordinates_detail:
+            coordinates.append(coordinate_detail['coor_set'])
+
+        num_coordinate = 1
+        for coordinate in coordinates:
+            (x,y) = coordinate
+            x = round(float(x))
+            x = x*10000
+            x = int(x)
+            x = str(x)
+            y = round(float(y))
+            y = y*10000
+            y = int(y)
+            y = str(y)
+            if(num_coordinate==1):
+                coordinate_soup = BeautifulSoup("<a:moveTo><a:pt x='"+x+"' y='"+y+"'/></a:moveTo>", 'xml')
+                path_tag.append(coordinate_soup.find('moveTo'))
+                num_coordinate+=1
+            else:
+                coordinate_soup = BeautifulSoup("<a:lnTo><a:pt x='"+x+"' y='"+y+"'/></a:lnTo>", 'xml')
+                path_tag.append(coordinate_soup.find('lnTo'))
+                num_coordinate+=1
+
+        #Adding color to the track
+        colors = track['color']
+        temp = colors.split("[")[1]
+        temp = temp[0:-1]
+        temp = temp.split(",")
+        r = int(temp[0].split("=")[1])
+        b = int(temp[1].split("=")[1])
+        g = int(temp[2].split("=")[1])
+
+        hex = "{:02x}{:02x}{:02x}".format(r,g,b)
```

Comment:
@purvil12c - I think ````hex```` is an existing Python object.  Could you use another name?

Reply:
Yes. Changing that.